flopy/mf6/data/mffileaccess.py

```diff
@@ -1355,7 +1355,9 @@ def read_list_data_from_file(
                             storage.data_dimensions.unlock()
                             return data_rec
             self.simple_line = (
-                self.simple_line and self.structure.package_type != "sfr"
+                self.simple_line
+                and self.structure.package_type != "sfr"
+                and self.structure.package_type != "uzf"
```

Comment:
I understand why this is here, but it doesn't seem like good practice to burn in the "sfr" and "uzf" names deep into low-level routines like this.  It would be great to come up with something more sustainable, but if that's too much for now, this is fine.

Reply:
@langevin-usgs, I agree that the package names should not be hard coded here.  I can work around this, but no solution I can think of at the moment is both easy and completely clean.  In order to understand why, let me give you some quick background on the problem, an example of the problem, and then a possible solution along with the potential problems with that solution.  

## Reason Behind Originally Hard Coding the SFR and UZF Names
The checks for the sfr and uzf packages are in the low level routines because it is the low level routines that determine whether or not to run the fast-loading code, for a simple recarray that has consistent header names and types, or the slower loading code, that handles more complex recarrays with inconsistent header names/types.  There is code in mffileaccess.py that attempts to generally detect which recarrays are simple and which complex.  However, there were some cases where this turned out to be difficult to detect.

For example, the sfr package’s packagedata block can have a cellid that is “NONE”.  There is currently no support for defining this in the dfn’s, so flopy does not know there is a potential “NONE” lurking in the cellid field until it encounters the first "NONE".  If the first “NONE” is far into the dataset, flopy will already have committed to the simple loading method and crash trying to load "NONE" as an integer.   

## Possible Temporary Solution to the Problem
Since, in general, flopy should not load recarrays using the simple method for any of the advanced packages, what I can do for now is add some code to mfstructure.py that determines whether the package is advanced, and then replace the checks for the sfr and uzf package with a more generalized check for any advanced packages.  Does this sound reasonable?

## Information the Proposed Solution Currently Lacks, and Ways to Work Around This
Note that there is one minor problem with the above suggested implementation, we do not currently explicitly define which packages are advanced in the dfns.  I see two options here, one is to add an “advanced package” flag at the top of each advanced package dfns, something like:

    # flopy advanced_package

The other option is to assume that any package with both a “packagedata” and “period” block is an advanced package.  This currently is the case, but as future packages are developed this generalization may or may not hold.  However, when the MF6 interface gets cleaned up it will likely take care of these problems, so perhaps this simple generalization that currently solves the problem is the best approach for now.

Reply:
Thanks for the explanation, @spaulins-usgs.  I think your simple proposal is fine for now, or even just leave it as is until the entire interface gets revamped.  It's up to you.  I just wanted to point out that we need to try and avoid this hard coding, especially as we are looking to implement more models and packages.